content/what-to-avoid.md

```diff
@@ -0,0 +1,32 @@
+# What to avoid
+
+The idea for this page came up during a discussion: quick reference page of
+things that you should not do with Git. Basically, a list of some common Git
+gotchas.
+
+**Postponing commits because the changes are "unfinished"/"ugly"**: It is
+better to have many OK-ish commits than too few perfect commits.  Too few
+perfect commits are probably too large and make undoing more difficult.  It is
+also easier to combine too small commits than it is to split too large commits.
+
+**Commit unrelated changes together**: Makes it difficult to undo changes since
+it can undo also the unrelated change.
+
+**Too ambitious branch which risks to never get completed**: The branch will
+never merge back and be so big and so ambitious with too many/big features that
+the risk is high that once it is really ready, there are conflicts everywhere.
+
+**Committing generated files**: See {ref}`gitignore`.
+
+**Over-engineering the branch layout and safeguards in small projects**: This
+may prevent people from contributing. Add more restrictions and safeguards as
+the project and the group of collaborators grows.
+
+**Commit messages what has been changed but do not explain why it has been
+changed**: This is as useful as code comments which describe the "obvious" such
+as "this is a loop" instead of explaining why something is done this way.
```

Comment:
```suggestion
as "this is a loop" instead of explaining why something is done this way.
But don't let perfect commit messages stop you from the most important point, committing often (first point).
```